apps/rewards/src/invite/inviteReward.service.ts

```diff
@@ -0,0 +1,171 @@
+import { KomenciLoggerService } from '@app/komenci-logger'
+import { networkConfig, NetworkConfig } from '@app/utils/config/network.config'
+import { isAccountConsideredVerified } from '@celo/base/lib'
+import { CeloContract, ContractKit } from '@celo/contractkit'
+import { Inject, Injectable } from '@nestjs/common'
+import { Cron, CronExpression } from '@nestjs/schedule'
+import { Raw } from 'typeorm'
+import { v4 as uuidv4 } from 'uuid'
+import { NotifiedBlock } from '../blocks/notifiedBlock.entity'
+import { NotifiedBlockRepository } from '../blocks/notifiedBlock.repository'
+import { InviteReward, RewardStatus } from './inviteReward.entity'
+import { InviteRewardRepository } from './inviteReward.repository'
+
+const WEEKLY_INVITE_LIMIT = 20
+
+@Injectable()
+export class InviteRewardService {
+  isRunning = false
+  komenciAddresses = []
+
+  constructor(
+    private readonly inviteRewardRepository: InviteRewardRepository,
+    private readonly notifiedBlockRepository: NotifiedBlockRepository,
+    private readonly contractKit: ContractKit,
+    @Inject(networkConfig.KEY)
+    private readonly networkCfg: NetworkConfig,
+    private readonly logger: KomenciLoggerService
+  ) {
+    this.komenciAddresses = this.networkCfg.relayers.map(
+      relayer => relayer.externalAccount
+    )
+  }
+
+  @Cron(CronExpression.EVERY_MINUTE)
+  async sendInviteRewards() {
+    if (this.isRunning) {
+      this.logger.log('Skipping because previous run is still ongoing')
+      return
+    }
+    this.isRunning = true
+
+    const cUsdTokenAddress = (
+      await this.contractKit.registry.addressFor(CeloContract.StableToken)
+    ).toLowerCase()
+
+    await this.usingLastNotifiedBlock(async fromBlock => {
+      this.logger.log(
+        `Starting to fetch invite rewards from block ${fromBlock}`
+      )
+      const escrow = await this.contractKit.contracts.getEscrow()
+      const withdrawalEvents = await escrow.getPastEvents('Withdrawal', {
```

Comment:
I think we should have a max block range for the query to help paginate the results and limit the load on Forno. This is less important for day-to-day running, but useful in the beginning or if we ever need to "catch up" after some time down.